### PR TITLE
[icu] mingw support + [mp3lame] Fix VCPKG_TARGET_STATIC_LIBRARY_SUFFIX

### PR DESCRIPTION
**Pull request by longnguyen2004**
**Describe the pull request**

- What does your PR fix?
Add support for `icu` on mingw triplets, and also add support for building `boost-locale` with `icu`
- Which triplets are supported/not supported? Have you updated the CI baseline?
All mingw triplets
`boost-locale[icu]` tested on both `x64-mingw-dynamic` and `x64-windows`
- Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
Yes

### CODE REVIEW

**Comment by PhoebeHui** [comment]
@longnguyen2004, the icu:x64-windows-static failed with following issue in CI testing, could you please confirm if this issue reprod locally?

Failures:
```
-- Package x64-windows-static-dbg
-- Package x64-windows-static-dbg done
CMake Error at ports/icu/portfile.cmake:237 (file):
  file RENAME failed to rename
    D:/packages/icu_x64-windows-static/lib/sicudt..lib
  to
    D:/packages/icu_x64-windows-static/lib/icudt..lib

  because: No such file or directory

Call Stack (most recent call first):
  scripts/ports.cmake:136 (include)
Error: Building package icu:x64-windows-static failed with: BUILD_FAILED
```

**Comment by longnguyen2004** [comment]
Guess I'll have to fix `mp3lame` as well, since there was a similar problem.

**Review by PhoebeHui — APPROVED** [review]

**Review by PhoebeHui — APPROVED** [review]
LGTM, thanks for the PR!

File: ports/boost-locale/CONTROL [hunk 1]
```diff
@@ -1,8 +1,8 @@
 # Automatically generated by scripts/boost/generate-ports.ps1
```
Comment (ras0219-msft) [inline ports/boost-locale/CONTROL:1]: This change needs to be added to `generate-ports.ps1`

**Review by ras0219-msft — CHANGES_REQUESTED** [review]

**Comment by BillyONeal** [comment]
Is the `-Bsymbolic` change submitted upstream to ICU or is there some reason it should not be?

**Comment by longnguyen2004** [comment]
I haven't submitted a patch to ICU yet, but I'll look into it.
FWIW, `-Bsymbolic` is only meaningful for ELF platforms, and lld also doesn't support it in the mingw driver, so I think upstream will accept it.

File: ports/boost-locale/CONTROL (hunk 1 above)
Comment (longnguyen2004) [reply ports/boost-locale/CONTROL:1]: Done

**Comment by BillyONeal** [comment]
> I haven't submitted a patch to ICU yet, but I'll look into it.
> FWIW, `-Bsymbolic` is only meaningful for ELF platforms, and lld also doesn't support it in the mingw driver, so I think upstream will accept it.

Can you actually submit it to them?

Thanks!

**Comment by longnguyen2004** [comment]
Submitted:
Jira ticket: https://unicode-org.atlassian.net/browse/ICU-21435
https://github.com/unicode-org/icu/pull/1505

**Comment by longnguyen2004** [comment]
@BillyONeal Seems like the fix will be left to 69.1, which is a long time away, can you check this PR again and make sure it's good?

**Comment by BillyONeal** [comment]
Absolutely: backports are fair game in patches! We just want to make sure upstream knows a bug exists etc. before we backport them in patches.

**Review by BillyONeal — APPROVED** [review]

**Review by ras0219-msft — CHANGES_REQUESTED** [review]
The generated port version of `boost-locale` must also be changed:
https://github.com/microsoft/vcpkg/pull/14969/files#diff-1fc3b5ebed025c0311a4625fa496b6db0580e6a7e97a88d0f9fbf402cd83d571R28

**Comment by BillyONeal** [comment]
I'm taking care of that, stand by

**Review by ras0219-msft — APPROVED** [review]

**Comment by BillyONeal** [comment]
Thanks for your contribution!

**Pull request merged**